File: backend/transaction-service/transaction-domain/transaction-application-service/src/main/java/com/saver/system/transaction/service/domain/ports/output/repository/TransactionRepository.java
package com.saver.system.transaction.service.domain.ports.output.repository;

import com.saver.system.domain.valueobject.UserId;
import com.saver.system.transaction.service.domain.entity.Transaction;
import java.util.Optional;

public interface TransactionRepository {
    Transaction save(Transaction transaction);
    Optional<Transaction> findByUserId(UserId userId);
}

File: backend/Services/transactions/src/data-access/repository/transaction-repository.ts
import { injectable } from "inversify";
import { DataSource, Repository } from "typeorm";
import { ITransactionRepository } from "../../domain/domain-application-service/ports/output/repository/itransaction-repository";
import { CashTransaction } from "../../domain/domain-core/entity/cash-transaction";
import { CashTransactionEntity } from "../entity/cash-transaction";

@injectable()
export class TransactionRepository extends Repository<CashTransactionEntity> implements ITransactionRepository {
    constructor(private dataSource: DataSource) {
        super(CashTransactionEntity, dataSource.createEntityManager());
    }

    async createCashTransaction(transaction: CashTransaction) {
        
    }
}